scripts/formatTest.js

```diff
@@ -0,0 +1,46 @@
+const { spawn } = require('child_process')
+
+const testScript = process.argv.find((arg, i, array) => array[i - 1] === '-t')
+const formatter = process.argv.find((arg, i, array) => array[i - 1] === '-with')
+
+runTestsWithFormatter(testScript, formatter)
+
+function runTestsWithFormatter (testScript, formatter = './node_modules/.bin/tap-spec') {
+  if (!testScript) {
+    console.log('No test script specified!')
+    return
+  }
+
+  const npmTestScriptNames = [
+    'coverageTests',
+    'testVM',
+    'testStateByzantium',
+    'testStateConstantinople',
+    'testBuildIntegrity',
+    'testBlockchain',
+    'testBlockchainGeneralStateTests',
+    'testBlockchainBlockGasLimit',
+    'testBlockchainValid',
+    'testBlockchainTotalDifficulty',
+    'testAPI',
+    'test'
+  ]
```

Comment:
Is this really necessary to duplicate the script names here or could this be done in a more general way? Would we loose much if we just allow this for every command and if it is not there it just breaks?

Reply:
Hi @danjm, can you give a short opinion on the two questions above?

Reply:
It is there so that we can distinguish which tests need to be run with `npm run` and which with `node`.

One approach we could take is to read the npm scripts dynamically.

Reply:
d9b4ea9 changes `scripts/formatTest.js` so that it reads directly from `package.json`. This removes the need to hard code any script names and make any changes to the file if/when `package.json` is changed.